spec/app/ics-020-fungible-token-transfer/README.md

```diff
@@ -256,27 +320,58 @@ function sendFungibleTokens(
 
 ```typescript
 function onRecvPacket(packet: Packet) {
-  FungibleTokenPacketData data = packet.data
+  channel = provableStore.get(channelPath(portIdentifier, channelIdentifier))
+  if channel.version === "ics20-1" {
+      FungibleTokenPacketData data = packet.data
+      trace, denom = parseICS20V1Denom(data.denom)
+      token = Token{
+        denom: denom
+        trace: trace
+        amount: packet.amount
+      }
+      tokens = []Token{token}
+  } else if channel.version === "ics20-2" {
+    FungibleTokenPacketDataV2 data = packet.data
+    tokens = data.tokens
+  }
```

Comment:
should be add an `abortUnless(channel.version == "ics20-1" || channel.version == "ics20-2")?

Reply:
Doesn't `onRecvPacket` operate within the context of an already established IBC channel? If so, then the `channel.version` through which these packets are received should have been previously set up and agreed upon to be either `ics20-1` or `ics20-2`.

Reply:
we have those assumptions, but I think it would still be preferable to be explicit in showing these are the only 2 possible states we can be in.

Also a similar point that this shouldn't be referencing `channel.version` since `channel.version` is not going to be the same as `application version` due to the structure of the version string when dealing with middleware.